mscore/scoreview.cpp

```diff
@@ -1229,6 +1229,19 @@ void ScoreView::resizeEvent(QResizeEvent* /*ev*/)
       {
       if (_magIdx != MagIdx::MAG_FREE)
             setMag(mscore->getMag(this));
+
+      int dx = 0;
+      int dy = 0;
+
+      constraintCanvas(&dx, &dy);
```

Comment:
Can you comment this code explaining the intended effect?  Right now it's the zoom behavior that is most concerning to me. and I sense this is meant to address it somehow, but it isn't clear how.

Reply:
This code was added to keep the score centered in the score view when the score view is resized. Of course, it will only do that if the option is checked, and the score is zoomed out enough that two opposite edges are visible.

Reply:
It does it by panning the score with a dx and dy of 0, which is to say not at all, and then scrolling by the amount necessary to keep the score centered. I basically copied the code from the end of ScoreView::zoom().

Reply:
Those who need zooming to stay the way it is will not want to enable this option. Those who are willing to make the trade-off will quickly learn how to deal with how zooming is affected. It is rather predictable, after all. If zoomed out far enough, the score will always be centered in the score view. Zooming back in will keep the score centered until it fills the score view. At that point, you will need to re-focus the zoom "hotspot" before zooming in again.